docs/source/user-guide/tasks/build-packages/compiler-tools.rst

```diff
@@ -338,3 +338,103 @@ This example is a little contrived, because the ``m2w64-gcc_win-64`` package is
+the runtime requirements but is not actually used in any of the exes or DSOs. This would allow us to consider
+these as candidates for removal thus making the package less heavy-weight. For the curious, there's a function
+in ``conda-build`` called ``check_overlinking`` and the original intention for this function was to do this but
+it ended up checking for the opposite (worse!) problem of 'underlinking'. This we define as missing
+dependencies, which happens frequently when something gets installed in the ``host prefix`` through a
+transitive dependency and is therefore not listed as a direct run dependency despite being directly linked to
+some exe(s) and/or DSO(s) in the package (build systems are sometimes 'greedy' linking to whatever they can
+find). We denote this as 'worse' because when someone installs these two packages (3 including the dependency)
+and then removes (and cleans/prunes the environment) the one that directly depends upon the third package, the
+third package will also be removed because conda has no idea your package needs it and it will no longer load.
+
+Unfortunately in most linkers the flag used to enable this (``--as-needed``) only takes effect for libraries
+that appear after it on the command-line. We have patched our ``libtool`` package (with a modified patch
+sourced from Gentoo) so this flag, if found, is moved to before any libraries in this linker command-line.
+The patch can be found at https://github.com/AnacondaRecipes/libtool-feedstock/blob/master/recipe/0001-link-as-needed.patch
+
+To ensure we take advantage of this flag, it is necessary to run ``autoreconf -vfi`` so that the ``libtool``
+script is regenerated. Before running ``configure``. For this you will need to add some dependencies to
+``meta.yaml``: ``libtool``, ``autoconf``, ``automake``, ``make`` and occasionallty for good measure:
+``pkg-config`` and ``bison``, depending on whether the package needs them or not.
+
+RPATHs (specifically rpath-link)
+--------------------------------
+
+On Linux, transitive DSOs are not searched for by the static linker in any `rpath` entries. Instead, for the
+linker to find these DSOs, it needs prompting via adding `-Wl,-rpath-link,$PREFIX/lib` to `LDFLAGS`. In the
+future we may add this to the Linux compiler activation script. These transitive DSOs are not added as `DT_NEEDED`
+entries in the top-level DSO, theya re inspected by the static linker for symbol resolution purposes only so it
+is generally safe to add this flag.
+
+CDT packages
+------------
+
+`CDT` packages are an 'Anaconda Distribution' invention. They consist of repackaged 'CentOS6' packages for libraries
+such as the `X11` and `OpenGL` stacks, unpacked into our compilers inbuilt sysroot. These allow us to link to these
+system libraries in an way that pins the compatibility level to that of 'CentOS6'. 'conda-forge' provide their own
+`X11` libraries and what to do with `OpenGL` is currently a subject of discussion.
```

Comment:
double `` escape in X11 & OpenGL?

Reply:
I was going for ``'s when referring to conda stuff (so CDTs I definitely agree) and '' for system stuff, *but* I'm not sure it's too important either way. If you have time, please do a PR.

Reply:
ok, makes sense to separate emphasis and keyword. I would double-tick keywords as well.